File: app/src/main/java/com/efraespada/stringobfuscator/MainActivity.java
package com.efraespada.stringobfuscator;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.TextView;

import com.stringcare.library.SC;
import com.stringcare.library.SCTextView;
import com.stringcare.library.Version;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        SC.init(getApplicationContext());

        String password = getString(R.string.snake_msg_hidden);
        String original = SC.reveal(password, Version.V3);

        String message = "Snake, the password is " + password + original;

        ((TextView) findViewById(R.id.programmatically_obfuscation)).setText(message);

        String numbers = getString(R.string.pattern, "hi", 3) + " is " + SC.reveal(R.string.pattern, "hi", 3);
        ((TextView) findViewById(R.id.pattern)).setText(numbers);

        final SCTextView tvAuto = findViewById(R.id.auto_tv);
        findViewById(R.id.btn_change).setOnClickListener(v -> {
            if (tvAuto.isHtmlEnabled()) {
                tvAuto.setHtmlSupport(!tvAuto.isHtmlEnabled());
            } else if (tvAuto.isRevealingValue()) {
                tvAuto.setRevealed(!tvAuto.isRevealingValue());
            } else if (!tvAuto.isRevealingValue()) {
                tvAuto.setRevealed(!tvAuto.isRevealingValue());
                tvAuto.setHtmlSupport(!tvAuto.isHtmlEnabled());
            }
        });

        boolean equals = SC.reveal(R.string.hello_world_b).equals(getString(R.string.hello_world_a));
        String areEquals = "Same result: " + equals;
        ((TextView) findViewById(R.id.same_value)).setText(areEquals);

        String jsonObjectName = SC.reveal(R.string.asset_json_file);
        SC.asset().asyncJson(jsonObjectName, json -> ((TextView) findViewById(R.id.json_object)).setText(json.toString()));
        SC.asset().asyncBytes(jsonObjectName, bytes -> ((TextView) findViewById(R.id.json_object_original)).setText(new String(bytes)), false);

        String jsonArrayName = SC.reveal(R.string.asset_json_raw_file);
        SC.asset().asyncJsonArray(jsonArrayName, json -> ((TextView) findViewById(R.id.json_array)).setText(json.toString()));
        SC.asset().asyncBytes(jsonArrayName, bytes -> ((TextView) findViewById(R.id.json_array_original)).setText(new String(bytes)), false);

    }
}

File: app/src/main/java/com/efraespada/stringobfuscator/MainActivity.kt
package com.efraespada.stringobfuscator

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.stringcare.library.*
import com.stringcare.library.SC.Companion.init
import com.stringcare.library.SC.Companion.reveal

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        init(applicationContext)
        val password = getString(R.string.snake_msg_hidden)
        val original = reveal(password, Version.V3)
        val message = "Snake, the password is $password$original"
        (findViewById<View>(R.id.programmatically_obfuscation) as TextView).text = message
        val numbers =
            getString(R.string.pattern, "hi", 3) + " is " + reveal(R.string.pattern, "hi", 3)
        (findViewById<View>(R.id.pattern) as TextView).text = numbers
        val tvAuto = findViewById<SCTextView>(R.id.auto_tv)
        findViewById<View>(R.id.btn_change).setOnClickListener { v: View? ->
            if (tvAuto.isHtmlEnabled) {
                tvAuto.setHtmlSupport(!tvAuto.isHtmlEnabled)
            } else if (tvAuto.isRevealingValue) {
                tvAuto.setRevealed(!tvAuto.isRevealingValue)
            } else if (!tvAuto.isRevealingValue) {
                tvAuto.setRevealed(!tvAuto.isRevealingValue)
                tvAuto.setHtmlSupport(!tvAuto.isHtmlEnabled)
            }
        }
        val equals = reveal(R.string.hello_world_b) == getString(R.string.hello_world_a)
        val areEquals = "Same result: $equals"
        (findViewById<View>(R.id.same_value) as TextView).text = areEquals
        val jsonObjectName = R.string.asset_json_file.reveal()


        findViewById<TextView>(R.id.json_object).text = jsonObjectName.json().toString()
        findViewById<TextView>(R.id.json_object_original).text =
            String(jsonObjectName.bytes { false })


        val jsonArrayName = R.string.asset_json_raw_file.reveal()
        findViewById<TextView>(R.id.json_array).text = jsonArrayName.jsonArray().toString()
        findViewById<TextView>(R.id.json_array_original).text =
            jsonArrayName.bytes { false }.toString()

    }
}